pkg/services/ngalert/models/alertmanager.go

```diff
@@ -27,4 +33,29 @@ type SaveAlertmanagerConfigurationCmd struct {
 	ConfigurationVersion      string
 	Default                   bool
 	OrgID                     int64
+	LastApplied               int64
+}
+
+// MarkConfigurationAsAppliedCmd is the command for marking a previously saved configuration as successfully applied.
+type MarkConfigurationAsAppliedCmd struct {
+	OrgID             int64
+	ConfigurationHash string
+}
+
+// GetAppliedConfigurationsQuery is the query for getting configurations that have been previously applied with no errors.
+type GetAppliedConfigurationsQuery struct {
+	OrgID  int64
+	Result []*HistoricAlertConfiguration
+}
+
+func HistoricConfigFromAlertConfig(config AlertConfiguration) HistoricAlertConfiguration {
+	return HistoricAlertConfiguration{
+		AlertConfiguration: AlertConfiguration{
+			AlertmanagerConfiguration: config.AlertmanagerConfiguration,
+			ConfigurationHash:         config.ConfigurationHash,
+			ConfigurationVersion:      config.ConfigurationVersion,
```

Comment:
Missing CreatedAt?

Reply:
Wouldn't that be automatically created by the DB at insert time?

Reply:
Maybe, is that name special to xorm? I honestly don't know

I also was thinking, we could just preserve the value from the main table, it's a reasonable value IMO (the timestamp that the config was created at, as opposed to the timestamp the history row was created at)